### DISCUSSION TITLE
Chat/correspondence table

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5760] [Feature Requests]
>One of our collections had a feature in their old system to be able to record conversations and link them to an object (in this case a loan). Are there any plans in the future for Specify to have a table that can store comments and/or conversations on a collection object/preparation/loan etc.?

**Requested By:** Dan Baker at CSIRO ([Asana](https://app.asana.com/0/1201972579963460/1204179977408329/f))

**Comment by grantfitzsimmons** [comment]
**Dan B:**
> I was imaging a simple text system rather than an attachment system since it would be easier to search through. In another ELN-type program I've used there was a small "comment" icon on an experiment to let users write some text - either some instructions to a user who might get it transferred to, or just a general note about the validity of the data.
> 
> For Specify I think it would be nice to have this on any of the main tables (CO/CE/Det/Prep/Loan) so that users can add a comment and see the history of previous comments. Our example is from the insect collection where they would add conversations about a loan - things like requests for extensions and any email exchange text with couriers etc.
> 
> Another useful addition would be the option to tag another user so they get a notification like you can do here in Asana

**Comment by maxpatiiuk** [comment]
A better solution might be to improve social media share previews of specify links.

Example use case:
1. The organization uses slack or etc for internal communication - and we don't have to reinvent a chat system
2. When someone sends a link to a collection object, the URL preview would have the formatted name of the specimen, and the image would come from the first attachment
3. Slack other provides a powerful search functionality

Otherwise, if they need something simple, perhaps it can be accomplished by the remarks field?
Or should we create a form plugin that stores conversation history in the remarks field?

**Comment by grantfitzsimmons** [comment]
How would you deal with Specify installations on private networks? Security concerns with making the first image public? Most of the time this information is behind a username and password. It is not feasible for many institutions, especially ones that are security-conscious

**Comment by danb213** [comment]
I agree that these comments don't always want to be public - there are examples where we need "internal" notes just for users of the system that don't get exported out.

I like the idea of co-opting the remarks fields in the tables - perhaps a configuration setting to turn on a "remarks history" so you can see the historical user/date/remarks. You would need a way to view that history on a form and search through the details - in this case doing a LIKE search on the aggregated remarks would be very useful (#2870)

**Comment by grantfitzsimmons** [comment]
**From Dan:**

> I was imaging a simple text system rather than an attachment system since it would be easier to search through. In another ELN-type program I've used there was a small "comment" icon on an experiment to let users write some text - either some instructions to a user who might get it transferred to, or just a general note about the validity of the data.
> 
> For Specify I think it would be nice to have this on any of the main tables (CO/CE/Det/Prep/Loan) so that users can add a comment and see the history of previous comments. Our example is from the insect collection where they would add conversations about a loan - things like requests for extensions and any email exchange text with couriers etc.
> 
> Another useful addition would be the option to tag another user so they get a notification like you can do here in Asana

**Comment by g1rly-c0d3r** [comment]
linked in https://github.com/specify/specify7/issues/8210
